scripts/ci/logcheck/blocklist-patterns

```diff
@@ -11,3 +11,49 @@ critical
 data race
 should (not|never) happen
 OMITTED.*LOG LINES DUE TO THROTTLING
+
+# Following is the list of deprecated k8s and OCP APIs that are getting signalled in the product logs by k8s client
+# libraries.
+# Obtain the list of deprecations:
+# - for k8s from https://kubernetes.io/docs/reference/using-api/deprecation-guide/
+# - for OCP via links from https://access.redhat.com/articles/6955985
```

Comment:
Per this thread <https://redhat-internal.slack.com/archives/CE4UC6AVB/p1685117395078889>, OCP does not yet have an established process of **own** APIs deprecation, e.g. the ones from `*.openshift.io` group.

One example of a deprecated API at the moment is `ImageContentSourcePolicy` and it does not seem to be used in ACS yet.

Reply:
This is the one https://github.com/stackrox/stackrox/pull/5007/files#diff-b8882059abee28a28ddc31d9ceecbd057f38642cf731cc31ed71fec4ece918faR66 but OCP APIs need to implement certain Go interface for API server to emit `Warning` header and client library to log it. My understanding is that it may not be implemented yet.

Note that all k8s deprecations listed above in the file will also currently match that `W[\d\s\.:]+warnings\.go:\d+].+` regex but I prefer to have them listed still because the actual logging call can be moved outside of `warnings.go` without a prior notice.